src/hotspot/share/opto/idealGraphPrinter.cpp

```diff
@@ -657,8 +653,8 @@ void IdealGraphPrinter::visit_node(Node* n, bool edges) {
         assert(typeLong->is_con(), "must be constant");
         jlong value = typeLong->get_con();
 
-        // max. 2 chars allowed
-        if (value >= -9 && value <= 99) {
+        // Only use up to 4 chars and fall back to a generic "L" to keep it short.
+        if (value >= -999 && value <= 9999) {
```

Comment:
Just for the sake of saying something, can we somewhat factor this test? And ideally, it could be marginally nicer, if I want to see 6 chars, to just replace a 4 into a 6 and it'd work magically. But I don't really see a way beside length(to_string(value)) <= max_length, and it's not really efficient so... If you have a brilliant idea, good. Otherwise, feel free to ignore: changing it as it is now is easy too.

Reply:
Good point! I think we can directly check how many chars are written with `snprintf_checked` and use that as an indicator. I pushed an update.